nautobot/core/celery/task.py

```diff
@@ -4,12 +4,47 @@
 from celery.result import EagerResult
 from celery.utils.functional import maybe_list
 from celery.utils.nodenames import gethostname
+from django.conf import settings
+from django.contrib.auth import get_user_model
 from kombu.utils.uuid import uuid
 
+from nautobot.core.branching import BranchContext
+
 
 class NautobotTask(Task):
     """Nautobot extensions to tasks for integrating with Job machinery."""
 
+    def before_start(self, task_id, args, kwargs):
+        User = get_user_model()
+        user_id = self.request.properties.get("nautobot_job_user_id", None)
+        self._nautobot_branch_context = BranchContext(
+            branch_name=self.request.properties.get("nautobot_job_branch_name", None),
+            user=User.objects.get(id=user_id) if user_id else None,
+        )
+        self._nautobot_branch_context.__enter__()
+
+    def after_return(self, status, retval, task_id, args, kwargs, einfo):
+        self._nautobot_branch_context.__exit__(None, None, None)  # TODO: good enough?
+        delattr(self, "_nautobot_branch_context")
```

Comment:
Why is this necessary?

Reply:
The delattr shouldn't be strictly necessary, I was just cleaning up after myself.

I should check the TODO still to see whether there's any inherent value to translating `retval`/`einfo` to the equivalent `exc_type`, `exc_value`, `traceback` that context manager `__exit__` can receive. Right now the branch context doesn't do anything with those params but it might be useful as future-proofing?